cmd/fyne/internal/commands/package.go

```diff
@@ -412,6 +421,35 @@ func mergeMetadata(p *appData, data *metadata.FyneApp) {
 	}
 }
 
+// normaliseIcon takes a non-png image file and converts it to PNG for use in packaging.
+// Successful conversion will return a path to the new file.
+// Any errors that occur will be returned with an empty string for new path.
+func normaliseIcon(path string) (string, error) {
+	// convert icon
+	img, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to open source image: %w", err)
+	}
+	defer img.Close()
+	srcImg, _, err := image.Decode(img)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode source image: %w", err)
+	}
+
+	tmpPath := filepath.Join(os.TempDir(), "fyne-ico-tmp.png")
+	out, err := os.Create(tmpPath)
```

Comment:
How about using [ioutil.TempFile()](https://pkg.go.dev/io/ioutil#TempFile) instead?
```suggestion
	out, err := ioutil.TempFile("", "fyne-ico-tmp.png")
```

Reply:
I used that originally, but it's not in Go 1.14

Reply:
Are you sure about that? It seems to exist in Go 1.14 if you look at this: https://pkg.go.dev/io/ioutil@go1.14#TempFile. The `os.CreateTemp` stuff was added in Go 1.16 but this does the same thing.